app/jobs/demo_mode/account_generation_job.rb

```
@@ -8,9 +8,15 @@ def perform(session, **options)
         raise "Unknown persona: #{session.persona_name}" if persona.blank?
 
         signinable = persona.generate!(variant: session.variant, password: session.signinable_password, options: options)
-        session.update!(signinable: signinable)
+        if signinable.present?
+          session.update!(signinable: signinable, status: 'successful')
+        else
+          raise "Failed to create signinable persona!"
+        end
+      rescue StandardError => e
+        session.update!(status: 'failed')
+        Rails.logger.error(e)
```

Sorry, last minute thought. Is there a reason we don't want to re-raise the exception here?

The benefits of re-raising would be:
* Alerts (e.g. Sentry)
* Applications could configure retries, discarded exceptions, etc
* The exceptions would be logged anyway

---

I had thought of this previously but because of the way the rescue was previously nested within the `lock` updating the `session` when rescuing an error was messy, but now it's pretty simple. So yeah i think this is the right approach.